### PR TITLE
feat(detail): add groups configuration redirection uri

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Add redirection for groups configuration

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Jira details
## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Infrastructure/RealTime/Api/Hypermedia/HypermediaService.php [hunk 1]
```diff
@@ -72,4 +72,18 @@ public function createInternalUris(mixed $response): array
         }
         return [];
     }
+
+    /**
+     * @param mixed $response
+     * @return array<array<string, string|null>>
+     */
+    public function createInternalGroupsUri(mixed $response): array
```
Comment (callapa) [inline src/Core/Infrastructure/RealTime/Api/Hypermedia/HypermediaService.php:80]: No information on the role of the method

File: src/Core/Infrastructure/RealTime/Api/Hypermedia/MetaServiceHyperMediaProvider.php [hunk 2]
```diff
@@ -188,4 +188,12 @@ public function createInternalUris(mixed $response): array
             'reporting' => $this->createForReporting($parameters),
         ];
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function createInternalGroupsUri(mixed $response): array
```
Comment (callapa) [inline src/Core/Infrastructure/RealTime/Api/Hypermedia/MetaServiceHyperMediaProvider.php:195]: No information on the role of the method

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by callapa — APPROVED** [review]

File: src/Core/Infrastructure/RealTime/Api/Hypermedia/HostHypermediaProvider.php [hunk 3]
```diff
@@ -138,4 +160,23 @@ public function createForTimelineEndpoint(array $parameters): string
...
+
+    /**
+     * @inheritDoc
+     */
+    public function createInternalGroupsUri(mixed $response): array
+    {
+        if (empty($response->hostgroups)) {
+            return [];
+        }
+
```
Comment (kduret) [inline src/Core/Infrastructure/RealTime/Api/Hypermedia/HostHypermediaProvider.php:172]:
```suggestion
```

File: src/Core/Infrastructure/RealTime/Api/Hypermedia/ServiceHyperMediaProvider.php [hunk 4]
```diff
@@ -183,4 +184,44 @@ public function createInternalUris(mixed $response): array
...
+
+    /**
+     * @inheritDoc
+     */
+    public function createInternalGroupsUri(mixed $response): array
+    {
+        if (empty($response->servicegroups)) {
+            return [];
+        }
+
```
Comment (kduret) [inline src/Core/Infrastructure/RealTime/Api/Hypermedia/ServiceHyperMediaProvider.php:217]:
```suggestion
```

**Review by kduret — APPROVED** [review]

**Pull request merged**